cache/cache_test.cc

```diff
@@ -734,17 +734,36 @@ TEST_P(CacheTest, ApplyToAllCacheEntiresTest) {
 TEST_P(CacheTest, DefaultShardBits) {
   // test1: set the flag to false. Insert more keys than capacity. See if they
   // all go through.
-  std::shared_ptr<Cache> cache = NewCache(16 * 1024L * 1024L);
-  ShardedCache* sc = dynamic_cast<ShardedCache*>(cache.get());
-  ASSERT_EQ(5, sc->GetNumShardBits());
+  std::shared_ptr<Cache> cache;
+  ShardedCache* sc;
 
-  cache = NewLRUCache(511 * 1024L, -1, true);
+  // 8 MiB
+  cache = NewCache(size_t{8} << 20);
   sc = dynamic_cast<ShardedCache*>(cache.get());
-  ASSERT_EQ(0, sc->GetNumShardBits());
+  EXPECT_EQ(4, sc->GetNumShardBits());
 
-  cache = NewLRUCache(1024L * 1024L * 1024L, -1, true);
+  // 16 MiB
+  cache = NewCache(size_t{16} << 20);
   sc = dynamic_cast<ShardedCache*>(cache.get());
-  ASSERT_EQ(6, sc->GetNumShardBits());
+  EXPECT_EQ(5, sc->GetNumShardBits());
+
+  // 511 KiB (used to be too small for > 1 shard, now considered ok)
```

Comment:
Too small shard size would cause over capacity problems. Single or a few large pinged blocks will make block size greatly oversized. We frequently have large blocks, e.g. index and filter blocks. So I would still cap single shard size not to be too small. Most server workloads use GBs of cache anyway, so it shouldn't impact workloads we care about.